CMakeLists.txt

```diff
@@ -251,7 +251,7 @@ if (USE_STATIC_LIBRARIES AND HAVE_NO_PIE)
     set (CMAKE_C_FLAGS                   "${CMAKE_C_FLAGS} ${FLAG_NO_PIE}")
 endif ()
 
-if (NOT SANITIZE)
+if (NOT SANITIZE AND NOT SPLIT_SHARED_LIBRARIES)
```

Comment:
Don't understand, CC @abyss7

Reply:
without this many libs in contrib is broken (not linked to pthread, m, ....)